### PR TITLE
Feature/speed up get connected devices

### PR DESCRIPTION
**Pull request by CoqRogue**
Speeds up getConnectedDevices by only returning serialnumbers and not information requiring to connect to the probe. It is possible to turn this on and off.

### CODE REVIEW

File: doc/api.js [hunk 1]
```diff
@@ -208,9 +208,17 @@
...
+ * @property {module:pc-nrfjprog-js~DeviceInformation} deviceInfo=undefined
+ * @property {module:pc-nrfjprog-js~ProbeInformation} probeInfo=undefined
+ * @property {module:pc-nrfjprog-js~LibraryInformation} libraryInfo=undefined
+ */
+
+/**
+ * Option flags to be used when getting serialnumbers.
+ * @typedef GetConnectedDevicesOptions
+ * @property {boolean} onlyGetSerialnumbers=false
+ *    Select wheter getConnectedDevices shall fetch DeviceInformation, ProbeInformation, and LibraryInformation
```
Comment (bencefr) [inline doc/api.js]: s/wheter/whether/

**Comment by bencefr** [comment]
I don't want to complicate this, but wouldn't it be cleaner to have a `getConnectedSerialNumbers()`? Just an idea.

**Comment by mrodem** [comment]
I agree with @bencefr that a dedicated `getSerialNumbers` function might be better. Now the developer must deal with an extra boolean flag, and the output will be `SerialNumberAndDeviceInformation` objects (ref. the doc) with only a serial number, which is not that intuitive.

Getting connected serial numbers will probably be needed in many cases, as `program`, `read`, `write` etc. all use the serial number to identify which device to use. So I think we should try to make this operation as simple as possible.

**Comment by CoqRogue** [comment]
Agreed. Will create a separate function.

File: doc/api.js [hunk 2]
```diff
@@ -298,7 +298,21 @@ export function getDllVersion(callback) {}
  */
 export function getConnectedDevices(callback) {}
 
-
+/**
+ * Async function to get a list of all connected devices serialnumber.
```
Comment (mrodem) [inline doc/api.js]: Perhaps reword to something like "... get the serial numbers of all connected devices"?

File: doc/api.js [hunk 3]
```diff
@@ -298,7 +298,21 @@ export function getDllVersion(callback) {}
...
+ *
+ * @example
+ * nrfjprogjs.getSerialNumbers(function(err, serialNumbers) {
+ *      if (err) throw err;
+ *      for (let i = 0; i < devices.length; i++) {
+ *          console.log(devices[i]);
+ *      }
```
Comment (mrodem) [inline doc/api.js]: s/devices/serialNumbers/

Comment (CoqRogue) [reply doc/api.js]: Fixed it when I copied it out to test, but forgot to fix in the source. Will fix.

**Pull request merged**